src/Http/Http.Extensions/src/ProblemDetailsJsonOptionsSetup.cs

```diff
@@ -13,13 +12,8 @@ public void PostConfigure(string? name, JsonOptions options)
     {
         if (options.SerializerOptions.TypeInfoResolver is not null)
```

Comment:
Why are we checking for null?

Reply:
I don't think make sense we setup our internal context when the app is configured for reflection-based, and actually this condition might be something like `options.SerializerOptions.TypeInfoResolver is not null && if (options.SerializerOptions.TypeInfoResolver is not DefaultTypeInfoResolver` or `options.SerializerOptions.TypeInfoResolver is JsonSerializerContext`

Reply:
Why?

Reply:
I feel like it is useless, I believe our context will never be used in this case since it will be added later or even worse, if it was `null`, now the app might break because we set a context. The only scenario where adding it helps it if, when `null`, another post-config sets a context.

Reply:
Why does the app break if we set a context?

Reply:
If you do something like this:

```c#
var options = httpContext.RequestServices.GetService<IOptions<JsonOptions>>.Value.SerializerOptions;
JsonSerializer.Serialize(new MyPoco(), options);
```

When it is `null` it will serialize using the reflection-based resolver, however, if we add when null it means our context will be used instead, and unless configured, will not fallback to reflection-based resolver and since we don't that `MyPoco` declared in our context it will fail.

Reply:
That seems broken, we need to think through this a bit more

Reply:
The summary is, once you add a context, the reflection based one will not be used ever?

Reply:
This breaks right:

```C#
var options = new JsonSerializerOptions();
options.AddContext<MyContext>();
JsonSerializer.Serialize(new Student("David", 1), options);

record Student(string Name, int Grade);
record Person(string Name);

[JsonSerializable(typeof(Person))]
partial class MyContext : JsonSerializerContext
{

}
```

Reply:
> The summary is, once you add a context, the reflection based one will not be used ever?

Correct. If the `TypeInfoResolver` property has been configured, then the serializer will not attempt to populate it with reflection resolution.

Reply:
> The summary is, once you add a context, the reflection based one will not be used ever?

The only scenario it will not be true is if you set the `DefaultTypeInfoResolver` first, but in this case your source gen context will not be used. This will work:

``` c#
var options = new JsonSerializerOptions() { TypeInfoResolver = new DefaultJsonTypeInfoResolver() };
options.AddContext<MyContext>();
JsonSerializer.Serialize(new Student("David", 1), options);

record Student(string Name, int Grade);
record Person(string Name);

[JsonSerializable(typeof(Person))]
partial class MyContext : JsonSerializerContext
{}
```

Reply:
Just be make it clear, the call to `AddContext` append the new context to be executed later.

https://github.com/dotnet/runtime/blame/2b8514b02f6d0e87f4645aca0be38f16864004a7/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs#L179

Since the reflection-based will be able to GetTypeInfo we will never call the source generated context

https://github.com/dotnet/runtime/blob/2b8514b02f6d0e87f4645aca0be38f16864004a7/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoResolver.cs#L69-L74